services/streammanager/googlepubsub/googlepubsubmanager.go

```diff
@@ -41,27 +54,36 @@ func NewProducer(destinationConfig interface{}) (*PubsubClient, error) {
 	if err != nil {
 		return nil, fmt.Errorf("[GooglePubSub] error  :: error in GooglePubSub while unmarshelling destination config:: %w", err)
 	}
-	var client *pubsub.Client
-	if config.Credentials != "" && config.ProjectId != "" {
-		client, err = pubsub.NewClient(ctx, config.ProjectId, option.WithCredentialsJSON([]byte(config.Credentials)))
+	if config.ProjectId == "" {
```

Comment:
in the previous implementation if `Credentials` or `ProjectId` were empty, we were ignoring it which would cause a panic in line 53 (nil pointer)